### PR TITLE
Adopt Swift 5.9

### PR DESCRIPTION
**Pull request by dfed**
As title. Swift 5.9 introduces new API that allow us to remove our force unwraps!

### CODE REVIEW

File: AsyncQueue.podspec [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
   s.name     = 'AsyncQueue'
-  s.version  = '0.4.0'
+  s.version  = '0.5.0'
   s.license  = 'MIT'
   s.summary  = 'A queue that enables ordered sending of events from synchronous to asynchronous code.'
   s.homepage = 'https://github.com/dfed/swift-async-queue'
   s.authors  = 'Dan Federman'
   s.source   = { :git => 'https://github.com/dfed/swift-async-queue.git', :tag => s.version }
-  s.swift_version = '5.8'
+  s.swift_version = '5.9'
```
Comment (bachand) [inline AsyncQueue.podspec:9]: 👍

File: Package.swift [hunk 2]
```diff
@@ -1,4 +1,4 @@
-// swift-tools-version: 5.8
+// swift-tools-version: 5.9
```
Comment (bachand) [inline Package.swift:1]: 👍

File: Sources/AsyncQueue/ActorQueue.swift [hunk 3]
```diff
@@ -57,13 +57,8 @@ public final class ActorQueue<ActorType: Actor>: @unchecked Sendable {
...
     /// Instantiates an actor queue.
     public init() {
-        var capturedTaskStreamContinuation: AsyncStream<ActorTask>.Continuation? = nil
-        let taskStream = AsyncStream<ActorTask> { continuation in
-            capturedTaskStreamContinuation = continuation
-        }
-        // Continuation will be captured during stream creation, so it is safe to force unwrap here.
-        // If this force-unwrap fails, something is fundamentally broken in the Swift runtime.
-        taskStreamContinuation = capturedTaskStreamContinuation!
+        let (taskStream, taskStreamContinuation) = AsyncStream<ActorTask>.makeStream()
```
Comment (bachand) [inline Sources/AsyncQueue/ActorQueue.swift:60]: 🎉

File: Gemfile.lock [hunk 4]
```diff
@@ -44,7 +51,7 @@ GEM
       public_suffix (~> 4.0)
       typhoeus (~> 1.0)
     cocoapods-deintegrate (1.0.5)
-    cocoapods-downloader (1.6.3)
+    cocoapods-downloader (2.1)
```
Comment (bachand) [inline Gemfile.lock:54]: This is a major version update. I am commenting so that this stands out.

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -16,45 +16,9 @@ jobs:
...
-    steps:
-      - name: Checkout Repo
-        uses: actions/checkout@v3
-      - name: Bundle Install
-        run: bundle install
-      - name: Select Xcode Version
-        run: sudo xcode-select --switch /Applications/Xcode_14.3.1.app/Contents/Developer
-      - name: Build and Test Framework
-        run: xcrun swift test -c release -Xswiftc -enable-testing
+        run: bundle exec pod lib lint --verbose --fail-fast --swift-version=5.9
```
Comment (bachand) [inline .github/workflows/ci.yml:21]: 👍

**Review by bachand — APPROVED** [review]
Nice!

File: Sources/AsyncQueue/ActorQueue.swift (hunk 3 above)
Comment (dfed) [reply Sources/AsyncQueue/ActorQueue.swift:60]: So. Much. Better!

**Pull request merged**